ev3dev2/motor.py

```diff
@@ -2099,6 +2111,178 @@ def follow_line(self,
+            as 0.01 (10ms).
+
+        ``follow_for`` is called to determine if we should keep following the
+            desired angle or stop.  This function will be passed ``self`` (the current
+            ``MoveWithGyroTank`` object). Current supported options are:
+            - ``follow_for_forever``
+            - ``follow_for_ms``
+
+        ``**kwargs`` will be passed to the ``follow_for`` function
+
+        Example:
+
+        .. code:: python
+
+            from ev3dev2.motor import OUTPUT_A, OUTPUT_B, MoveTank, SpeedPercent, follow_for_ms
+            from ev3dev2.sensor.lego import GyroSensor
+
+            # Instantiate the MoveTank object
+            tank = MoveTank(OUTPUT_A, OUTPUT_B)
+
+            # Initialize the tank's gyro sensor
+            tank.gyro = GyroSensor()
+
+            try:
+                # Calibrate the gyro to eliminate drift, and to initialize the current angle as 0
+                tank.calibrate_gyro()
+
+                # Follow the line for 4500ms
+                tank.follow_gyro_angle(
+                    kp=11.3, ki=0.05, kd=3.2,
+                    speed=SpeedPercent(30),
+                    target_angle=0
+                    follow_for=follow_for_ms,
+                    ms=4500
+                )
+            except FollowGyroAngleErrorTooFast:
+                tank.stop()
+                raise
+        """
+        assert self.gyro, "GyroSensor must be defined"
+
+        integral = 0.0
+        last_error = 0.0
+        derivative = 0.0
+        speed_native_units = speed.to_native_units(self.left_motor)
+        MAX_SPEED = SpeedNativeUnits(self.max_speed)
+
+        while follow_for(self, **kwargs):
+            current_angle = self.gyro.angle
+            error = current_angle - target_angle
+            integral = integral + error
+            derivative = error - last_error
+            last_error = error
+            turn_native_units = (kp * error) + (ki * integral) + (kd * derivative)
+
+            left_speed = SpeedNativeUnits(speed_native_units - turn_native_units)
+            right_speed = SpeedNativeUnits(speed_native_units + turn_native_units)
+
+            if left_speed > MAX_SPEED:
+                log.info("%s: left_speed %s is greater than MAX_SPEED %s" %
+                        (self, left_speed, MAX_SPEED))
+                self.stop()
+                raise FollowGyroAngleErrorTooFast(
+                    "The robot is moving too fast to follow the angle")
+
+            if right_speed > MAX_SPEED:
+                log.info("%s: right_speed %s is greater than MAX_SPEED %s" %
+                        (self, right_speed, MAX_SPEED))
+                self.stop()
+                raise FollowGyroAngleErrorTooFast(
+                    "The robot is moving too fast to follow the angle")
+
+            if sleep_time:
+                time.sleep(sleep_time)
+
+            self.on(left_speed, right_speed)
+
+        self.stop()
+
+    def pivot_gyro(self,
```

Comment:
How about `turn_to_angle_gryo`? That would be more consistent with `MoveDifferential.turn_to_angle()`

Reply:
Makes sense.